network/router.go

```diff
@@ -118,6 +118,17 @@ func (r *Router) Start() {
 			}
 			return
 		}
+
+		// Reject incoming connections from invalid peers
+		if !isPeerValid(dst.ID) {
+			log.Errorf("rejecting incoming connection from %v: invalid peer %v",
+				c.Remote(), dst.ID)
+			if err := c.Close(); err != nil {
+				log.Error("closing connection:", err)
+			}
+			return
```

Comment:
Why doesn't this return an error?

Reply:
Because `Start()` does not return one (called as a goroutine).

Reply:
But this runs in `Listen`, which does return an error, no?

Reply:
Indeed not `Start()`, but as I understand it, it actually runs in the function argument to `Listen()`, which does not return an error. What am I missing?

Reply:
I was really looking hard at the code and thought I'm probably wrong - but I didn't see it until now.  `r.host.Listen` returns an error, but not the function given in the method. Sorry for the noise...